runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java

```diff
@@ -0,0 +1,235 @@
+package net.runelite.client.plugins.musiclist;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.runelite.api.*;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.VarClientIntChanged;
+import net.runelite.api.widgets.*;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.game.chatbox.ChatboxPanelManager;
+import net.runelite.client.game.chatbox.ChatboxTextInput;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@PluginDescriptor(
+		name = "MusicListPlugin",
+		description = "Adds search and filter for the music list"
+)
+
+public class MusicListPlugin extends Plugin
+{
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private ChatboxPanelManager chatboxPanelManager;
+
+	private ChatboxTextInput searchInput;
+
+	private Widget musicSearchButton;
+	private Widget musicFilterButton;
+
+	private Collection<Widget> tracks;
+
+	private MusicState currentMusicFilter;
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged e)
+	{
+		if (e.getGameState() == GameState.LOGIN_SCREEN)
+		{
+			currentMusicFilter = MusicState.ALL;
+			tracks = null;
+		}
+	}
+
+	@Subscribe
+	public void onVarClientIntChanged(VarClientIntChanged varClientIntChanged)
+	{
+		if (varClientIntChanged.getIndex() == VarClientInt.INVENTORY_TAB.getIndex())

```

Comment:
Why can't this be created just when the music interface is loaded? Creating/destroying it each time the tab changes seems unnecessary.

Reply:
You're absolutely correct. I was unaware of the existence of the WidgetLoaded class prior, but I have fixed this issue.